flair/models/lemmatization_model.py

```diff
@@ -0,0 +1,516 @@
+            teacher_forcing_ratio: float = 0.5,
+            longest_word_length: int = 50
+
+    ):
+        """
+        :param hidden_size: number of hidden states in RNN
+        :param pre_embeddings: pre-trained embeddings
+        :param character_dictionary: dictionary containing all the characters in the corpus
+        :param n_layers: number of RNN layers
+        :param tag_list: list of tags planned to be used
+        :param tag_dictionary: dictionary of tags you want to predict
+        :param dropout: dictionary contains all the tag values in the tag list
+        :param teacher_forcing_ratio: the probability of using teacher_forcing. If it is greater than or equal to 1,
+        always use teacher forcing. If less than or equal to 0, never use.
+        :param longest_word_length: record the maximum length of words and lemma to use in prediction.
+        Based on experience, the default value is set at 50.
+        """
+        super(Lemmatization, self).__init__()
+        self.hidden_size = hidden_size
+        self.pre_embeddings = pre_embeddings.to(flair.device)
+        self.n_layers = n_layers
+        self.teacher_forcing_ratio = teacher_forcing_ratio
+        self.dropout = dropout
+        self.character_dictionary = character_dictionary
+        self.character_dictionary.add_item(SOS_token)
+        self.character_dictionary.add_item(EOS_token)
+
+        self.pre_embedding_weight = self._load_embedding_weight(self.character_dictionary, self.pre_embeddings)
+        self.embedding = nn.Embedding(len(self.character_dictionary), self.pre_embeddings.embedding_length)
+        self.embedding.weight.data.copy_(self.pre_embedding_weight)
+        self.embedding.weight.requires_grad = False
+        self.embedding_dropout = nn.Dropout(dropout)
+
+        # TODO: LSTM model implementation
+        self.encoder = nn.GRU(self.pre_embeddings.embedding_length,
+                              hidden_size,
+                              n_layers,
+                              dropout=(0 if n_layers == 1 else dropout),
+                              bidirectional=True)
+        self.decoder = nn.GRU(self.pre_embeddings.embedding_length,
+                              hidden_size,
+                              n_layers,
+                              dropout=(0 if n_layers == 1 else dropout))
+
+        self.concat = nn.Linear(self.hidden_size * 2, self.hidden_size)
+        self.out = nn.Linear(self.hidden_size, len(self.character_dictionary))
+
+        # If tag_list is not empty, it is considered that the user needs to use tag.
+        if len(tag_list):
+            self.use_tag: bool = True
+            self.tag_list = tag_list
+            self.tag_dictionary = tag_dictionary
+            self.tag_embeddings = nn.Embedding(len(self.tag_dictionary), self.pre_embeddings.embedding_length)
+            self.tag_embeddings.weight.data.uniform_(-1, 1)
+        else:
+            self.use_tag: bool = False
+
```

Comment:
`self.tag_list = tag_list` should be set in the else clause as well, otherwise we encounter is a problem when saving the model (`_get_state_dict` looks for `self.tag_list`).